ddl/ddl_api.go

```diff
@@ -3753,6 +3753,9 @@ func (d *ddl) TruncateTable(ctx sessionctx.Context, ti ast.Ident) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if tb.Meta().IsView() || tb.Meta().IsSequence() {
+		return infoschema.ErrTableNotExists.GenWithStackByArgs(schema.Name.O, tb.Meta().Name.O)
```

Comment:
`ErrTableNotExists` is not suitable.

Reply:
MySQL throws this error when truncating view.
MariaDB throws this error when truncating these two.